pxr/imaging/hdSt/materialXFilter.cpp

```diff
@@ -1264,8 +1193,10 @@ _GenerateMaterialXShader(
     mxHdInfo.bindlessTexturesEnabled = bindlessTexturesEnabled;
 
     // Generate the glslfx source code from the mtlxDoc
-    return HdSt_GenMaterialXShader(
+    mx::ShaderPtr mxShader = HdSt_GenMaterialXShader(
         mtlxDoc, stdLibraries, searchPaths, mxHdInfo, apiName);
+
+    return std::make_shared<HdSt_MaterialXCodegenResult>(*mxShader);
```

Comment:
Creating a codegen result from the MaterialX Shader, and destroying the latter because we've already extracted everything we will ever need from it.